packages/x-components/src/composables/use-store.ts

```diff
@@ -1,13 +1,33 @@
-import { getCurrentInstance } from 'vue';
+import { computed, ComputedRef, getCurrentInstance } from 'vue';
 import { Store } from 'vuex';
+import { ExtractState, XModuleName } from '../x-modules/x-modules.types';
 
 /**
- * Function which returns the `$store` object from the current component instance.
+ * Function which returns the `$store` object from the current component instance
+ * and the selected state.
  *
- * @returns The `$store` object from the current component instance.
+ * @param module - The {@link XModuleName} of the getter.
+ * @param path - The state path.
+ * @returns The state properties of the module and the `$store`.
  *
  * @public
  */
-export function useStore(): Store<any> {
-  return (getCurrentInstance()?.proxy as { $store: Store<any> }).$store;
+export function useStore<Module extends XModuleName, Path extends keyof ExtractState<Module>>(
+  module?: Module,
+  path?: Path
+): UseStore<Module, Path> {
+  const store = (getCurrentInstance()?.proxy as { $store: Store<any> }).$store;
+  const state = computed(() => store.state.x[module][path]);
+  return {
+    store,
+    state
+  };
```

Comment:
This would break if both `module` or `path` are not passed to the composable.
Instead of this approach, I'm thinking of something closer to what vueblocks is doing:
https://vueblocks.github.io/vue-use-utilities/guide/vuex/useVuex.html#example

For our case, we should return store always and the composable can receive the module name as optional.
Instead of returning the state directly, we return a `useState` function that receives the module and a list of paths to the state's properties. If the module name was provided when calling the `useStore` composable then it's binded to the `useState` function so it only needs the paths list